OpenRA.Game/Graphics/Renderable.cs

```diff
@@ -17,23 +17,26 @@ namespace OpenRA.Graphics
 	public interface IRenderable
 	{
 		WPos Pos { get; }
-		float Scale { get; }
 		PaletteReference Palette { get; }
 		int ZOffset { get; }
 		bool IsDecoration { get; }
 
-		IRenderable WithScale(float newScale);
 		IRenderable WithPalette(PaletteReference newPalette);
 		IRenderable WithZOffset(int newOffset);
 		IRenderable OffsetBy(WVec offset);
 		IRenderable AsDecoration();
 
-		void BeforeRender(WorldRenderer wr);
+		IFinalizedRenderable PrepareRender(WorldRenderer wr);
+	}
+
+	public interface IFinalizedRenderable
```

Comment:
This *is* the separate interface file, but I agree. I'll split `SpriteRenderable` out into its own file instead.